lib/services/jellyfin_api_helper.dart

```diff
@@ -655,4 +692,41 @@ class JellyfinApiHelper {
           if (maxHeight != null) "MaxHeight": maxHeight.toString(),
         });
   }
+
+  /// Returns the correct URL for the given item.
+  Uri getsongDownloadUrl({
+    required BaseItemDto item,
+    required DownloadProfile? transcodingProfile,
+  }) {
+    Uri uri = Uri.parse(_finampUserHelper.currentUser!.baseUrl);
+
+    if (transcodingProfile != null &&
+        transcodingProfile.codec != FinampTranscodingCodec.original) {
+      // uri.queryParameters is unmodifiable, so we copy the contents into a new
+      // map
+      final queryParameters = Map.of(uri.queryParameters);
+
+      // iOS/macOS doesn't support OPUS (except in CAF, which doesn't work from
+      // Jellyfin). Once https://github.com/jellyfin/jellyfin/pull/9192 lands,
```

Comment:
As per https://github.com/jmshrv/finamp/pull/496#issuecomment-1826930178, I think this is already released on 10.8

Reply:
As far as I can tell, only the fix to actually set the codec made it in to 10.8.x.  The fix to not backtrack in m4a containers seems to still be missing.

Reply:
Oh wow 😅